convert2rhel/unit_tests/special_cases_test.py

```diff
@@ -0,0 +1,80 @@
+from unittest import mock
+
+import pytest
+
+from convert2rhel import special_cases
+from convert2rhel.special_cases import (
+    OPENJDK_RPM_STATE_DIR,
+    perform_java_openjdk_workaround,
+)
+
+
+@pytest.mark.parametrize(
+    (
+        "has_openjdk",
+        "can_successfully_apply_workaround",
+        "mkdir_p_should_raise",
+        "check_message_in_log",
+        "check_message_not_in_log",
+    ),
+    [
+        # All is fine case
+        (
+            True,
+            True,
+            None,
+            "openjdk workaround applied successfully.",
+            "Can't create %s" % OPENJDK_RPM_STATE_DIR,
+        ),
+        # openjdk presented, but OSError when trying to apply workaround
+        (
+            True,
+            False,
+            OSError,
+            "Can't create %s" % OPENJDK_RPM_STATE_DIR,
+            "openjdk workaround applied successfully.",
+        ),
+        # No openjdk
+        (False, False, None, None, None),
+    ],
+)
+def test_perform_java_openjdk_workaround(
+    monkeypatch,
```

Comment:
Never heard of [`monkeypatch`](https://docs.pytest.org/en/stable/monkeypatch.html). You have it at the start and at the end of the parameters in the two functions, we should stick to having it in one position for consistency's sake

Reply:
I believe you meant fixtures? I put them at the start (monkeypatch) and at the end (caplog). ok, let's agree to put fixtures as last args:
```python
def test_perform_java_openjdk_workaround(
    has_openjdk,
    mkdir_p_should_raise,
    check_message_in_log,
    check_message_not_in_log,
    monkeypatch,
    caplog,
):
```